### ISSUE TITLE
[BUG]Logits shift to larger values over training

### ISSUE THREAD

**Issue by TissueC** [microsoft/DeepSpeed#4379]
**Describe the bug**
We found that in training llm, the logits (or softmax input) gradually become larger and larger over training.
For example, the logits_max of orginal LLaMa 7B/13B is around 10, but become very large to 1000+ after training.
The loss (or softmax output) seems very normal in training, but the generation degrading could be quite serious. The most evident thing is that the model becomes very sensitive to repetition_penalty, which is a common trick to control repetition by adjusting the logits.

Some additional experimental results to help locate and identify this issue:
1. lower learning rate (e.g. 2e-4->2e-5) can help slow down the speed of logits increasing.
2. larger batch size can help slow down the speed of logits increasing.
3. smaller number of nodes (gpu device) can help slow down the speed of logits increasing.
4. zero-2/3 all have this issue. No experiments on zero-0/1.
5. it seems that this issue occurs only when the number of GPU node is large and the model size is large.
6. This issue occurs in bf16 precision. No experiments in fp16.

The recent paper, [baichuan-2 technical report (p5, left-bottom)](https://cdn.baichuan-ai.com/paper/Baichuan2-technical-report.pdf) also mentions this issue.

[Issue 4195](https://github.com/microsoft/DeepSpeed/issues/4195) reports the same issue and that issue is closed. The issue mentions that the bug only occurs in DeepSpeed but not megatron.

Considering the wide impact of the DeepSpeed community and the large number of developers, I believe that many LLM training teams around the world have encountered this issue. I hope that this issue can be thoroughly discussed until it is completely resolved.

**Expected behavior**
The logits is stable over training.

**Comment by TissueC** [comment]
Hello, has there been any progress so far? @jomayeri
